activerecord/lib/active_record/fixtures.rb

```diff
@@ -1009,7 +1009,9 @@ def teardown_fixtures
         ActiveSupport::Notifications.unsubscribe(@connection_subscriber) if @connection_subscriber
         @fixture_connections.each do |connection|
           connection.rollback_transaction if connection.transaction_open?
-          connection.pool.lock_thread = false
+          connection.pool.lock_thread = false unless defined?(use_transactional_tests?) &&
+            use_transactional_test_case?
```

Comment:
This is needed so that lock_thread isn't disabled after each test, when it's required by the outer transaction. Perhaps a more flexible solution would to be to do something like move `lock_thread` to a property on each transaction, so that when a transaction is popped the outer property can be used? Could use some guidance on this.